File: scripts/seed.py
#!/usr/bin/env python3
"""
Quorum — Seed Script
Ingests Northwind dossier documents into local QVAC RAG vector store.
Usage: python3 scripts/seed.py
"""

import os
import sys
import json

DOSSIER_DIR = os.path.join(os.path.dirname(__file__), '..', 'data', 'fixtures', 'northwind_dossier')

def load_dossier():
    documents = []
    dossier_path = os.path.abspath(DOSSIER_DIR)
    if not os.path.isdir(dossier_path):
        print(f"[seed] ERROR: Dossier not found: {dossier_path}")
        sys.exit(1)
    for fname in sorted(os.listdir(dossier_path)):
        if fname.endswith('.txt'):
            fpath = os.path.join(dossier_path, fname)
            with open(fpath, 'r', encoding='utf-8') as f:
                content = f.read().strip()
            if content:
                documents.append({'filename': fname, 'content': content, 'chars': len(content)})
                print(f"  [seed] Loaded {fname} ({len(content)} chars)")
    return documents

def main():
    print("=" * 60)
    print("  Quorum — Seed Script")
    print("  Ingesting Northwind dossier into RAG vector store")
    print("=" * 60)
    print()
    documents = load_dossier()
    if not documents:
        print("[seed] ERROR: No documents found.")
        sys.exit(1)
    total_chars = sum(d['chars'] for d in documents)
    print(f"\n[seed] Total: {len(documents)} documents, {total_chars} chars")
    manifest = {
        'seeded_at': __import__('datetime').datetime.utcnow().isoformat() + 'Z',
        'documents': len(documents),
        'total_chars': total_chars,
        'files': [d['filename'] for d in documents],
        'planted_contradictions': [
            'VP Chen approved payment on March 12 BUT was on PTO March 11-15',
            'Memo REF-4821 timestamped March 12 BUT HR logs show no badge/VPN access',
            '$2.4M payment requires board resolution (Section 4.2) BUT no board discussion occurred',
        ],
    }
    manifest_path = os.path.join(os.path.dirname(__file__), '..', 'data', 'seed_manifest.json')
    os.makedirs(os.path.dirname(manifest_path), exist_ok=True)
    with open(manifest_path, 'w') as f:
        json.dump(manifest, f, indent=2)
    print(f"[seed] Manifest written: {os.path.abspath(manifest_path)}")
    print("\n[seed] ✅ Seed complete.")

if __name__ == '__main__':
    main()

File: scripts/seed.ts
/**
 * Seed Script — Ingest Northwind dossier into QVAC RAG vector store
 *
 * Usage: npx tsx scripts/seed.ts
 *
 * Reads all .txt files from data/fixtures/northwind_dossier/ and
 * ingests them into the local RAG corpus for the council to query.
 */

import { readFileSync, readdirSync } from "fs";
import { join } from "path";
import { close } from "@qvac/sdk";
import { ingestCorpus, resetCorpus, releaseEmbeddingModel } from "../src/core/rag.js";

const DOSSIER_DIR = join(process.cwd(), "data/fixtures/northwind_dossier");

async function main() {
  console.log("🏛️  Quorum — Corpus Seeder");
  console.log("─".repeat(50));

  const files = readdirSync(DOSSIER_DIR).filter((f) => f.endsWith(".txt"));
  if (files.length === 0) {
    console.error("❌ No .txt files found in", DOSSIER_DIR);
    process.exit(1);
  }

  console.log(`📂 Found ${files.length} documents in ${DOSSIER_DIR}:`);
  files.forEach((f) => console.log(`   • ${f}`));
  console.log();

  const documents = files.map((f) => {
    const content = readFileSync(join(DOSSIER_DIR, f), "utf-8");
    console.log(`📄 ${f} — ${content.length} chars`);
    return `[Source: ${f}]\n${content}`;
  });

  console.log(`\n⏳ Ingesting ${documents.length} documents into RAG corpus...`);
  const start = Date.now();

  await resetCorpus(); // idempotent re-seed: clear any prior corpus first
  await ingestCorpus(documents, true);

  const elapsed = Date.now() - start;
  console.log(`✅ Corpus ingestion complete in ${elapsed}ms`);

  // Release embedding model to free memory
  await releaseEmbeddingModel();
  console.log("🧹 Embedding model released");
  console.log("\n🏛️  Ready for council debates!");
}

main()
  .then(async () => {
    await close().catch(() => {}); // shut down the SDK worker so the process exits
    process.exit(0);
  })
  .catch(async (err) => {
    console.error("❌ Seed failed:", err);
    await close().catch(() => {});
    process.exit(1);
  });
